src/diffusion_for_multi_scale_molecular_dynamics/active_learning_loop/active_learning.py

```diff
@@ -0,0 +1,322 @@
+            uncertain_structure, uncertainty_per_atom = (
+                self._get_uncertain_structure_and_uncertainties(artn_working_directory)
+            )
+
+            number_of_uncertain_envs = np.sum(
+                uncertainty_per_atom > uncertainty_threshold
+            )
+            logger.info(
+                f" -> There are {number_of_uncertain_envs} environments with uncertainty above the threshold."
+            )
+
+            logger.info("  Making new samples based on uncertainties.")
+            list_sample_structures, list_sample_information = self._make_samples(
+                uncertain_structure, uncertainty_per_atom
+            )
+
+            logger.info("  Labelling samples with oracle...")
+            time1 = time.time()
+            list_single_point_calculations = [
+                self.oracle_single_point_calculator.calculate(structure)
+                for structure in list_sample_structures
+            ]
+            time2 = time.time()
+            logger.info(
+                f" -> It took {time2- time1: 6.2e} seconds to compute labels with Oracle."
+            )
+
+            logger.info("  Converting labelled samples and writing pickle to disk.")
+            oracle_df = self._convert_single_point_calculations_to_dataframe(
+                list_single_point_calculations, list_sample_information
+            )
+
+            oracle_directory = current_sub_directory / "oracle"
+            oracle_directory.mkdir(parents=True, exist_ok=True)
+
+            output_file = oracle_directory / "oracle_single_point_calculations.pkl"
+            oracle_df.to_pickle(output_file)
+
+            logger.info("  Adding samples and uncertain environment to FLARE.")
+            for single_point_calculation, sample_info in zip(list_single_point_calculations, list_sample_information):
+                active_environment_indices = self._determine_active_environment_indices(sample_info,
+                                                                                        uncertainty_threshold)
+                flare_trainer.add_labelled_structure(
+                    single_point_calculation,
+                    active_environment_indices=active_environment_indices,
+                )
+
+            logger.info("  Fitting the FLARE hyperparameters...")
+            optimization_result, history_df = flare_trainer.fit_hyperparameters()
+            logger.info(f"  Optimization status : {optimization_result.success}")
+            logger.info(f"  Optimization message : {optimization_result.message}")
+            logger.info("  The SGP hyperparameters are now : ")
+            sigma, sigma_e, sigma_f, sigma_s = optimization_result.x
+            logger.info(f"       sigma   = {sigma: 12.8f}")
+            logger.info(f"       sigma_e = {sigma_e: 12.8f}")
+            logger.info(f"       sigma_f = {sigma_f: 12.8f}")
+            logger.info(f"       sigma_s = {sigma_s: 12.8f}")
```

Comment:
all the parameters above are FLARE dependent. Maybe we could hide this logging in a flare_trainer method ? Or something like that. Could be done in another PR